prov/shm/src/smr_attr.c

```diff
@@ -80,8 +85,8 @@ struct fi_fabric_attr smr_fabric_attr = {
 };
 
 struct fi_info smr_info = {
-	.caps = FI_MSG | FI_SEND | FI_RECV | FI_SOURCE | FI_TAGGED | FI_RMA |
-		FI_READ | FI_WRITE | FI_REMOTE_READ | FI_REMOTE_WRITE | FI_ATOMICS,
+	.caps = SMR_TX_CAPS | SMR_RX_CAPS,
+	.mode = FI_CONTEXT,
```

Comment:
I don't see where this mode bit is needed.  Setting this requires that the app provide struct fi_context as its operation context parameter.  From what I can tell, the shm provider doesn't care what the user passes in as its context.   Nothing in this series needs it.

Reply:
The provider needs a valid context in the case of larger messages to track the pending command and report errors after finding a completed response. It also needs a context for receives that might be canceled (which I'm working on right now).

Reply:
I would drop this until the patch that needs it is added.  Adding mode bits reduces the number of apps that can use the provider.  In the worst case, we allocate a context internally if needed.